### PR TITLE
Excises the Developer guide section from the docs

### PR DESCRIPTION
**Pull request by MichaelDrogalis**
### Description 

For a while, we have been working towards disseminating the Developer guide section into the docs into both reference documentation and the new, more structured how-to guides. This patch deletes the last of it, judiciously redirecting to newer content.

### CODE REVIEW

File: docs/how-to-guides/test-an-app.md [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
-description: Test your SQL statements in ksqlDB without the need for a full Kafka cluster
----
-
-Use the ksqlDB testing tool to test a set of SQL statements. The ksqlDB
-testing tool is a command line utility that enables testing SQL
-statements without requiring any infrastructure, like {{ site.aktm }}
-and ksqlDB clusters. The ksqlDB testing tool is a great way to design your
-ksqlDB pipeline and ensure the expected results are generated. You can
-collaborate on designing your SQL statements by sharing the test files.
+# How to test an application
```
Comment (JimGalasyn) [inline docs/how-to-guides/test-an-app.md:1]: Maybe add the yaml front matter?

**Review by JimGalasyn — APPROVED** [review]
LGTM, with one suggestion.

File: docs/how-to-guides/test-an-app.md (hunk 1 above)
Comment (MichaelDrogalis) [reply docs/how-to-guides/test-an-app.md:8]: Good call, done

File: docs/how-to-guides/test-an-app.md [hunk 2]
```diff
@@ -86,23 +92,6 @@ the previous test:
 }
 ```
 
-#### Generating input files automagically from existing topic data
```
Comment (rmoff) [inline docs/how-to-guides/test-an-app.md:89]: @MichaelDrogalis Why was this dropped? Can it be reinstated somewhere else? It's useful information.

Comment (vcrfxia) [reply docs/how-to-guides/test-an-app.md:89]: https://github.com/confluentinc/ksql/pull/6826

**Pull request merged**